geo/src/algorithm/voronoi_diagram.rs

```diff
@@ -484,41 +513,46 @@ where
 fn get_inf_on_midpoint_triangle<T: GeoFloat>(
```

Comment:
this function is only used once, and IMO it should be split.

the `if midpoint == circumcentre` branch doesnt use `midpoint`
the `if midpoint != circumcentre` branch doesnt use `triangle` or `edge`

the `if midpoint == circumcentre` feels like it should be done by the caller (i.e. in `CircumCenterLocation::On => {...}`) , then each branch in the caller would be invoking functions that are quite different in nature and those function names can describe what they do, instead of being named by where they are called.

Reply:
I put `get_inf_on_midpoint_triangle` in a separate function so I could test it separately along with each of the infinity line construction methods, without the need to execute the larger function `define_edge_to_infinity`

Reply:
In test `test_inf_on_midpoint_triangle`, two of the three invocations of `get_inf_on_midpoint_triangle` are for the `circumcenter != midpoint` branch which is not interesting.  those two invocations are essentially tests for `get_inf_line_in_out_triangle`, not tests of the body of `get_inf_on_midpoint_triangle`.

Only one of the invocations is testing the much more interesting `midpoint == circumcentre` case.

If you really want to keep this function signature, IMO start it with an early-exit of the uninteresting case:

```rust
if circumcenter != midpoint {
    return Ok(get_inf_line_in_out_triangle(...)'
}
```

Reply:
I have moved the check to the top of the function.  IMO the distinction is based on the property of the triangle, which is how the functions are split up.  I agree that the interesting case is only 1/3 of the function calls, but I think it is easier to understand the process with the functions divided based on triangle property.

Reply:
I attempted to split it out and there are a lot of downstream code changes that would I think make the process a bit trickier to follow.